File: java/09_pgm_avion_interface/src/co/edu/campusucc/poo/interfaces/IAvionStandar.java
package co.edu.campusucc.poo.interfaces;

public interface IAvionStandar extends IAvionCarga, IAvionMilitar, IAvionGranja {

    void startingEngine();
    void stopingEngine();

    void speedUp();
    void currentSpeed();
    void landing();
    void descent();
    void chargeFuel(int litros);
    
    void setMarca(String marca);
    void setTipo(String tipo);

}

File: c++/09_pgm_avion_interface/src/co/edu/campusucc/poo/interfaces/IAvionStandar.cpp
#ifndef IAVIONSTANDAR_H
#define IAVIONSTANDAR_H

class IAvionCarga {
public:
    virtual void openDoors() = 0;
    virtual void closeDoors() = 0;
    virtual void pickUp() = 0;
};

class IAvionMilitar {
public:
    virtual void shootMissiles() = 0;
};

class IAvionGranja {
public:
    virtual void irrigation() = 0;
};

class IAvionStandar : public IAvionCarga, public IAvionMilitar, public IAvionGranja {
public:
    virtual void startingEngine() = 0;
    virtual void stopingEngine() = 0;
    virtual void speedUp() = 0;
    virtual void currentSpeed() = 0;
    virtual void landing() = 0;
    virtual void descent() = 0;
    virtual void chargeFuel(int litros) = 0;
    virtual void setMarca(const std::string& marca) = 0;
    virtual void setTipo(const std::string& tipo) = 0;
};

#endif
